models/evc.py

```diff
@@ -1410,6 +1410,45 @@ def handle_link_down(self):
 
         return success
 
+    @staticmethod
+    def get_interface_from_switch(uni: UNI, switches: dict) -> Interface:
+        """Get interface from switch by uni"""
+        switch = switches[uni.interface.switch.dpid]
+        interface = switch.interfaces[uni.interface.port_number]
+        return interface
+
+    def handle_topology_update(self, switches: dict):
+        """Handle changes in the topology"""
+        interface_a = self.get_interface_from_switch(self.uni_a, switches)
```

Comment:
probably won't happen, but it'd be great to handle `KeyError` at the call site here for `self.get_interface_from_switch` with a `log.warning` message and return early if it ever happens, topology send a copy of the topology, if on day we allow hard deletions then this would already be handled gracefully

Reply:
Added exception handlingsssssssssssssssssssssssssssss